azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/ChatMiddleware.kt

```diff
@@ -6,17 +6,39 @@ package com.azure.android.communication.ui.chat.redux.middleware
 import com.azure.android.communication.ui.chat.redux.Dispatch
 import com.azure.android.communication.ui.chat.redux.Middleware
 import com.azure.android.communication.ui.chat.redux.Store
-import com.azure.android.communication.ui.chat.redux.middleware.listener.ChatActionListener
+import com.azure.android.communication.ui.chat.redux.action.Action
+import com.azure.android.communication.ui.chat.redux.action.LifecycleAction
+import com.azure.android.communication.ui.chat.redux.middleware.listener.ChatActionHandler
+import com.azure.android.communication.ui.chat.redux.middleware.listener.ChatServiceListener
 import com.azure.android.communication.ui.chat.redux.state.ReduxState
+import com.azure.android.communication.ui.chat.service.ChatService
 
 internal interface ChatMiddleware
 
-internal class ChatMiddlewareImpl(
-    private val chatActionListener: ChatActionListener
+internal class ChatServiceMiddleware(
+    private val chatService: ChatService
 ) :
     Middleware<ReduxState>,
     ChatMiddleware {
-    override fun invoke(store: Store<ReduxState>): (next: Dispatch) -> Dispatch {
-        TODO("Not yet implemented")
+
+    private val chatActionHandler = ChatActionHandler(chatService = chatService)
+    private val chatServiceListener = ChatServiceListener(chatService = chatService)
+
+    override fun invoke(store: Store<ReduxState>) = { next: Dispatch ->
+        { action: Action ->
+
+            when (action) {
+                is LifecycleAction.Wakeup -> {
+                    chatService.init()
+                    chatServiceListener.startListening(store::dispatch)
```

Comment:
Chat service will probbaly always listen. We will start/stop notifications inside nativeSDK.